### ISSUE TITLE
Error while generating : undefined method `to_hash'

### ISSUE THREAD

**Issue by curly1875** [Kapeli/cheatset#40]
Hi,

Cheatset installation was successful but generating the provided example `sample.rb` fails.

# Versions used
macOS 12.6.2
ruby 3.2.0 (2022-12-25 revision a528908271) [x86_64-darwin21]
gem 3.4.1
```
$ gem which cheatset
~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/lib/cheatset.rb
$ gem which temple
~/.gem/ruby/3.2.0/gems/temple-0.9.1/lib/temple.rb
```

# Error
```
$ cheatset generate sample.rb
~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/lib/cheatset/dsl/base.rb:21: warning: undefining the allocator of T_DATA class Redcarpet::Markdown
~/.gem/ruby/3.2.0/gems/temple-0.9.1/lib/temple/map.rb:87:in `validate_map!': undefined method `to_hash' for "!!!\n%html\n  %head\n    %meta{charset: 'utf-8'}\n    %title= title\n    %link{rel: 'stylesheet', href: 'style.css'}\n    - if style\n      %style{type: 'text/css'}= style\n    - if head_inject\n      = head_inject\n    - categories.each do |category|\n      %link{href: \"//dash_ref/Category/\#{category.id.strip.gsub(/\\//, '%2F')}/1\"}\n      - category.hasEntry(false)\n      - category.entries.each_with_index do |entry, index|\n        - if entry.name || entry.index_name\n          - category.hasEntry(true)\n          %link{href: \"//dash_ref_\#{category.id.strip.gsub(/\\//, '%2F')}/Entry/\#{(entry.name) ? entry.tags_stripped_name.strip.gsub(/\\//, '%2F') : entry.index_name.strip.gsub(/\\//, '%2F')}/0\"}\n        - elsif entry.command && entry.command.length > 0\n          - category.hasEntry(true)\n          %link{href: \"//dash_ref_\#{category.id.strip.gsub(/\\//, '%2F')}/Command/\#{entry.command.first.strip.gsub(/\\//, '%2F')}/0\"}\n      - if !category.hasEntry\n        %link{href: \"//dash_ref/Entry/\#{category.id.strip.gsub(/\\//, '%2F')}/0\"}\n\n  %body\n    %header\n      %h1= title\n\n    %article\n      %p~ introduction\n\n      - categories.each do |category|\n        %section.category{class: category.html_class}\n          - if !category.hasEntry\n            %a{name: \"//dash_ref/Entry/\#{category.id.strip.gsub(/\\//, '%2F')}/0\"}\n          %h2{id:\"//dash_ref/Category/\#{category.id.strip.gsub(/\\//, '%2F')}/1\"}\n            = category.id\n          %div.scrollable\n            %table\n              - if category.header\n                %tr\n                  - category.header.each do |header|\n                    %th~ header\n\n              - category.entries.each_with_index do |entry, index|\n                %tr{id: ((entry.name || entry.index_name) ? \"//dash_ref_\#{category.id.strip.gsub(/\\//, '%2F')}/Entry/\#{(entry.name) ? entry.tags_stripped_name.strip.gsub(/\\//, '%2F') : entry.index_name.strip.gsub(/\\//, '%2F')}/0\" : \"//dash_ref_\#{category.id.strip.gsub(/\\//, '%2F')}/Command/\#{entry.command.first.strip.gsub(/\\//, '%2F')}/0\" if (entry.name || entry.index_name || (entry.command && entry.command.length > 0)) )}\n                  - if entry.command\n                    %td.command\n                      - entry.command.each do |command|\n                        %p\n                          %code= CGI.escapeHTML(command)\n                  - if entry.td_command\n                    - entry.td_command.each do |command|\n                      %td.td_command\n                        - if !command.empty?\n                          %code= CGI.escapeHTML(command)\n                  - if entry.name || entry.notes || entry.html_notes\n                    %td.description{:colspan => (\"2\" unless entry.command || entry.td_command || entry.td_notes)}\n                      .name~ entry.name\n                      .notes~ entry.notes\n                      .notes~ entry.html_notes\n                  - if entry.td_notes\n                    - entry.td_notes.each do |notes|\n                      %td.td_notes\n                        - if !notes.empty?\n                          .td_notes~ notes\n      - if notes\n        %section.notes\n          %h2 Notes\n          ~ notes\n    %footer\n      - if source_url\n        You can modify and improve this cheat sheet <a onclick=\"window.dash.openExternal_(this.href); return false;\" href=\"\#{source_url}\">here</a>\n      - else\n        Generated with <a onclick=\"window.dash.openExternal_(this.href); return false;\" href='https://github.com/Kapeli/cheatset#readme'>cheatset</a>":String (NoMethodError)

      map.to_hash.keys.each {|key| validate_key!(key) }
         ^^^^^^^^
Did you mean?  to_s
	from ~/.gem/ruby/3.2.0/gems/temple-0.9.1/lib/temple/mixins/options.rb:81:in `initialize'
	from ~/.gem/ruby/3.2.0/gems/temple-0.9.1/lib/temple/engine.rb:45:in `initialize'
	from ~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/lib/cheatset/creator.rb:34:in `new'
	from ~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/lib/cheatset/creator.rb:34:in `generate_html_file'
	from ~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/lib/cheatset/creator.rb:20:in `generate'
	from ~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/lib/cheatset/dsl/context.rb:9:in `generate'
	from ~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/lib/cheatset/cli.rb:27:in `generate'
	from ~/.gem/ruby/3.2.0/gems/thor-1.2.1/lib/thor/command.rb:27:in `run'
	from ~/.gem/ruby/3.2.0/gems/thor-1.2.1/lib/thor/invocation.rb:127:in `invoke_command'
	from ~/.gem/ruby/3.2.0/gems/thor-1.2.1/lib/thor.rb:392:in `dispatch'
	from ~/.gem/ruby/3.2.0/gems/thor-1.2.1/lib/thor/base.rb:485:in `start'
	from ~/.gem/ruby/3.2.0/gems/cheatset-1.4.6/bin/cheatset:6:in `<top (required)>'
	from ~/.gem/ruby/3.2.0/bin/cheatset:25:in `load'
	from ~/.gem/ruby/3.2.0/bin/cheatset:25:in `<main>'

```

Did I forget to install a dependency? Thank you for your help

**Comment by dickdavis** [comment]
Looks like this was caused by a [change](https://github.com/haml/haml/commit/519ef23b55cc5f8b29ecc9849006f272b3d78802) in the `haml` gem. I've got a PR up that resolves the issue.

**Comment by Smolations** [comment]
@Kapeli This seems like a really simple fix (pinning the `haml` gem to v5.2.2; and while you're at it, pin the other gems to be safe). It was reported almost a year ago. Why hasn't this been fixed?? 🤔
